draft-ietf-jsonpath-base.md

```diff
@@ -1069,7 +1069,10 @@ the current node is selected if and only if the expression yields true.
 
 The current node is accessible via the current node identifier `@`.
 
-An existence expression may test the result of a function expression (see {{fnex}}).
+An existence expression either tests the existence of a node
+designated by an embedded query (see {{extest}}) or tests whether the
+result of a function expression of type `Boolean` or `OptionalBoolean`
+(see {{fnex}}) is `true`.
```

Comment:
I don't think of functions expressions of type `Boolean` as existence expressions. We test the return value, but existence is moot. I'd prefer to raise them to another kind of basic-expr.

`OptionalBoolean` function expressions involve both an existence test and, if the return value is not `Nothing`, consideration of the return value.

I wonder if we should try to tease this out in the spec?

Reply:
I'd like to fix the ABNF to be structured along more logical lines.  This is really #344; I think in this PR we should make sure we have all the material that we then reshuffle.